### ISSUE TITLE
Regex backreferencing between word and substitution

### ISSUE THREAD

**Issue by uthbees** [richardfrost/AdvancedProfanityFilter#361]
#### :loudspeaker: The Shortcoming
I'm working on making replacement words make more sense in their context. Just now, I was trying to change "piss[ed] * off" to "tick[ed] * off", so I put `piss([ed]{0,2}[ ]?\w*) off` into word/phrase and `tick\1 off` into substitution. However it looks like APF doesn't support backreferencing between these fields, so instead of replacing, say, "piss John off" with "tick John off", it uses "tick\1 off".

#### :microscope: Describe the solution you'd like
My request is to add support for backreferencing between the Word/Phrase and Substitution fields, so that you can capture a group in the first field and backreference it in the second. This would allow for more dynamic filtering in cases such as I outlined above.

**Comment by richardfrost** [comment]
Currently the extension doesn't support backreferencing on the substitutions. It is something I could look at adding. The main reason why I haven't is it has seemed like almost nobody uses the regex mode, and its been "good" enough without it.

Using existing functionality right now, the most common ways to solve the issue with the example you provided would be either to add the word using the `Partial` match method, or to add variations (with and without `ed` at the end, etc.). Based on your example, I feel like those existing options would probably be sufficient, but if you have an example where you need to backreference it I'm open to adding it to the backlog.

**Comment by uthbees** [comment]
It's true that it's good enough without, but I think this would still be nice to have. My example in this case is that the middle word can be him, her, them, it, or any name (although admittedly I haven't actually noticed anyone using a name there). Currently I'm just replacing everything with "ticked 'em off", which isn't as easy to follow.

That being said, I will concede that this may not be worth your time as it's such a small improvement and very few people would likely use it.

**Comment by richardfrost** [comment]
I think that referenced commit should do the trick. I'll get it out in the next release and you can test it and let me know how it works for you. Thanks for the suggestion, and the detailed use-case for it.

**Comment by richardfrost** [comment]
I think that referenced commit should do the trick. I'll get it out in the next release and you can test it and let me know how it works for you. Thanks for the suggestion, and the detailed use-case for it @uthbees.

**Comment by richardfrost** [comment]
This was released in version [2.32.0](https://github.com/richardfrost/AdvancedProfanityFilter/releases/tag/2.32.0). Let me know how it works once you get a chance to test it @uthbees.

**Comment by uthbees** [comment]
TLDR: It seems to work in Chrome but not Firefox, but if you can't reproduce those results, I'll look into it some more.

Details: I added a new word to my word list with `piss([ed]{0,2}[ ]?\w*[ ]?[of]{0,3})`, `tick\1`, and regex, keeping all other settings at the default. This worked in Chrome, but not Firefox - in Firefox, it's just replacing things with "tick/1", like before. Since Firefox is what I've been using for a while and I just installed the extension in Chrome for this purpose, it's possible that there's some option I messed with that's causing the problem. However, I both exported my config from Firefox to Chrome and restored the default settings in Firefox and my results stayed the same, so I'm pretty confident it's just a browser issue. (I also tried starting from the default config in Firefox and deleting all the default words, leaving just the one entry, but that didn't fix it either.)

**Comment by uthbees** [comment]
After quite a bit more troubleshooting, I reinstalled the APF extension and it seems to work fine now! I exported/imported my normal config, so that should be the same as before. I'm not sure how to reproduce this bug, since to my knowledge there's no way to install an older version of an addon and upgrade from it. However, at least when installing the addon fresh, it looks good to me!

**Comment by richardfrost** [comment]
@uthbees Thanks for digging into the issue a bit more. I think if its working okay for you know, I'll go ahead and close this. If an issue comes up with it, feel free to re-open this issue or create a new one. Thanks again for the detailed suggestion!
